x-pack/legacy/plugins/reporting/server/routes/lib/make_request_facade.ts

```diff
@@ -4,28 +4,25 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { RequestQuery } from 'hapi';
-import { Legacy } from 'kibana';
+import { KibanaRequest, IBasePath, RequestHandlerContext } from 'src/core/server';
 import {
   RequestFacade,
   ReportingRequestPayload,
-  ReportingRequestPre,
   ReportingRequestQuery,
 } from '../../../server/types';
 
-export function makeRequestFacade(request: Legacy.Request): RequestFacade {
-  // This condition is for unit tests
-  const getSavedObjectsClient = request.getSavedObjectsClient
-    ? request.getSavedObjectsClient.bind(request)
-    : request.getSavedObjectsClient;
+export function makeRequestFacade(
+  context: RequestHandlerContext,
+  request: KibanaRequest,
+  basePath: IBasePath['get']
+): RequestFacade {
   return {
-    getSavedObjectsClient,
-    headers: request.headers,
-    params: request.params,
-    payload: (request.payload as object) as ReportingRequestPayload,
-    query: ((request.query as RequestQuery) as object) as ReportingRequestQuery,
-    pre: (request.pre as Record<string, any>) as ReportingRequestPre,
-    getBasePath: request.getBasePath,
+    getSavedObjectsClient: () => context.core.savedObjects.client,
```

Comment:
Ideally `getSavedObjectsClient` would be removed from the facade and passed down as other services at some point.